pkg/operator/certrotation/target.go

```diff
@@ -102,6 +104,36 @@ func needNewTargetCertKeyPair(annotations map[string]string, caBundleCerts []*x5
 	return true
 }
 
+func needNewTargetCertKeyPairForTime(annotations map[string]string, signer *crypto.CA, validity time.Duration, renewalPercentage float32) bool {
+	targetExpiry := annotations[CertificateExpiryAnnotation]
+	if len(targetExpiry) == 0 {
+		return true
+	}
+	certExpiry, err := time.Parse(time.RFC3339, targetExpiry)
+	if err != nil {
+		glog.Infof("bad expiry: %q", targetExpiry)
+		// just create a new one
+		return true
+	}
+
+	// If Certificate is past its validity, we may must generate new.
+	if time.Now().After(certExpiry) {
+		return true
+	}
+
+	// If Certificate is past its validity*renewpercent, we may have action to take. if the signer is old enough
+	renewalDuration := -1 * float32(validity) * (1 - renewalPercentage)
+	if time.Now().After(certExpiry.Add(time.Duration(renewalDuration))) {
+		// make sure the signer is at 10% or renewal overlap old so we had time to rotate trust. Before signing new.
+		timeToWaitForTrustRotation := -1 * renewalDuration / 10
+		if time.Now().After(signer.Config.Certs[0].NotBefore.Add(time.Duration(timeToWaitForTrustRotation))) {
```

Comment:
this requires that the signer renewal period is > 10% of cert validity. Should we document that? Maybe even check in the constructor (or whereever it makes sense)?